lib/solidus_social/engine.rb

```diff
@@ -16,6 +16,16 @@ class Engine < Rails::Engine
       Spree::SocialConfig = Spree::SocialConfiguration.new
     end
 
+    initializer 'solidus_social.decorate_spree_user' do
+      Rails.application.reloader.after_class_unload do
+        # Reload and decorate the spree user class immediately after it is
+        # unloaded so that it is available to devise when loading routes
+        if Rails.configuration.cache_classes == false
```

Comment:
Haha right ... learn 2 ruby.

I also realized when patching this into our app directly that that check is rather redundant, since the after_class_unload callback is only run when caching is disabled anyway